crates/types/src/services/p2p.rs

```diff
@@ -43,6 +43,19 @@ pub type TransactionGossipData = GossipData<Transaction>;
 /// Newly produced block notification
 pub type BlockGossipData = GossipData<Block>;
 
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+/// The source of some network data.
+pub struct SourcePeer<T> {
+    /// The source of the data.
+    pub peer_id: PeerId,
+    /// The data.
+    pub data: T,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+/// Opaque peer identifier.
+pub struct PeerId(Vec<u8>);
```

Comment:
We should make a tech-debt cleanup task for updating the other gossip data types here to use this new PeerId type.